document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutorTest.java

```diff
@@ -0,0 +1,306 @@
+                    ConstantExpression.ofStrings(List.of("Soap", "Bottle"))))
+            .build();
+
+    Query query = Query.builder().setSelection(selection).setFilter(filter).build();
+
+    Iterator<Document> resultDocs = collection.find(query);
+    assertDocsEqual(resultDocs, "mongo/simple_filter_response.json");
+  }
+
+  @Test
+  public void testFindWithSortingAndPagination() throws IOException {
+    List<SelectionSpec> selectionSpecs =
+        List.of(
+            SelectionSpec.of(IdentifierExpression.of("item")),
+            SelectionSpec.of(IdentifierExpression.of("price")),
+            SelectionSpec.of(IdentifierExpression.of("quantity")),
+            SelectionSpec.of(IdentifierExpression.of("date")));
+    Selection selection = Selection.builder().selectionSpecs(selectionSpecs).build();
+
+    Filter filter =
+        Filter.builder()
+            .expression(
+                RelationalExpression.of(
+                    IdentifierExpression.of("item"),
+                    IN,
+                    ConstantExpression.ofStrings(List.of("Mirror", "Comb", "Shampoo", "Bottle"))))
+            .build();
+
+    Sort sort =
+        Sort.builder()
+            .sortingSpec(SortingSpec.of(IdentifierExpression.of("quantity"), DESC))
+            .sortingSpec(SortingSpec.of(IdentifierExpression.of("item"), ASC))
+            .build();
+
+    Pagination pagination = Pagination.builder().offset(1).limit(3).build();
+
+    Query query =
+        Query.builder()
+            .setSelection(selection)
+            .setFilter(filter)
+            .setSort(sort)
+            .setPagination(pagination)
+            .build();
+
+    Iterator<Document> resultDocs = collection.find(query);
+    assertDocsEqual(resultDocs, "mongo/filter_with_sorting_and_pagination_response.json");
+  }
+
+  @Test
+  public void testFindWithNestedFields() throws IOException {
+    List<SelectionSpec> selectionSpecs =
+        List.of(
+            SelectionSpec.of(IdentifierExpression.of("item")),
+            SelectionSpec.of(IdentifierExpression.of("price")),
+            SelectionSpec.of(IdentifierExpression.of("props.seller.name"), "seller"));
+
+    Query query =
+        Query.builder()
+            .addSelections(selectionSpecs)
+            .setFilter(
+                LogicalExpression.builder()
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("item"),
+                            IN,
+                            ConstantExpression.ofStrings(List.of("Mirror", "Comb", "Shampoo"))))
+                    .operator(OR)
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("props.seller.address.pincode"),
+                            EQ,
+                            ConstantExpression.of(700007)))
+                    .build())
+            .addSort(IdentifierExpression.of("props.brand"), ASC)
```

Comment:
nit: `props.brand` and `props.seller.address.city` should have been also added in selection spec, so it's easy to see the order result for comparision.

Reply:
Done. Will push the changes.